localization/ndt_scan_matcher/src/ndt_scan_matcher_core.cpp

```diff
@@ -97,31 +97,32 @@ NDTScanMatcher::NDTScanMatcher()
   inversion_vector_threshold_(-0.9),
   oscillation_threshold_(10),
   output_pose_covariance_(),
-  regularization_enabled_(declare_parameter("regularization_enabled", false)),
+  regularization_enabled_(declare_parameter<bool>("regularization_enabled")),
   estimate_scores_for_degrounded_scan_(
-    declare_parameter("estimate_scores_for_degrounded_scan", false)),
-  z_margin_for_ground_removal_(declare_parameter("z_margin_for_ground_removal", 0.8))
+    declare_parameter<bool>("estimate_scores_for_degrounded_scan")),
+  z_margin_for_ground_removal_(declare_parameter<double>("z_margin_for_ground_removal"))
 {
   (*state_ptr_)["state"] = "Initializing";
   is_activated_ = false;
 
-  int points_queue_size =
-    static_cast<int>(this->declare_parameter("input_sensor_points_queue_size", 0));
+  int points_queue_size = this->declare_parameter<int>("input_sensor_points_queue_size");
   points_queue_size = std::max(points_queue_size, 0);
   RCLCPP_INFO(get_logger(), "points_queue_size: %d", points_queue_size);
 
-  base_frame_ = this->declare_parameter("base_frame", base_frame_);
+  base_frame_ = this->declare_parameter<std::string>("base_frame");
   RCLCPP_INFO(get_logger(), "base_frame_id: %s", base_frame_.c_str());
 
-  ndt_base_frame_ = this->declare_parameter("ndt_base_frame", ndt_base_frame_);
+  if (this->has_parameter("ndt_base_frame")) {
+    ndt_base_frame_ = this->declare_parameter<std::string>("ndt_base_frame");
```

Comment:
@TaikiYamada4 Please add `ndt_base_frame` to the yaml config. 
And remove if statement.